src/mobile/src/main/java/tupperdate/android/home/Home.kt

```diff
@@ -1,98 +1,146 @@
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material.Button
+import android.util.Log
+import androidx.compose.foundation.gestures.draggable
+import androidx.compose.foundation.layout.*
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.WithConstraints
+import androidx.compose.ui.gesture.scrollorientationlocking.Orientation
+import androidx.compose.ui.platform.DensityAmbient
+import androidx.compose.ui.platform.LifecycleOwnerAmbient
 import androidx.compose.ui.unit.dp
 import androidx.ui.tooling.preview.Preview
+import tupperdate.android.appbars.mainBottomBar
+import tupperdate.android.appbars.mainTopBar
 import tupperdate.android.ui.TupperdateTheme
-import tupperdate.api.AuthenticationApi
+import tupperdate.api.RecipeApi
+import tupperdate.api.api
+import kotlin.math.absoluteValue
 
 @Composable
+/*
+@ Guy-Laurent : I am forced to do only one function because it is the only way
+to call like() and dislike() without the weird RealRecipeApi import.
+ */
 fun Home(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile?,
-    modifier: Modifier = Modifier,
+    recipeApi: RecipeApi,
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    if (user == null) {
-        HomeDisconnected(
-            onButtonClick,
-            modifier,
-        )
-    } else {
-        HomeConnected(
-            onButtonClick,
-            user,
-            modifier,
-        )
-    }
-}
+    // TODO make a call to front element of recipeApi
+    val presentRecipe = (RecipeApi.Recipe(
+        "Red lobster",
+        "Look at me, am I not yummy ?",
+        "lobster.jpg"
+    ))
 
-@Composable
-private fun HomeDisconnected(
-    onButtonClick: () -> Unit,
-    modifier: Modifier,
-) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there.")
-        Button(onClick = onButtonClick) {
-            Text("Go to onboarding")
-        }
-    }
+    mainTopBar(onChatClick = onChatClick, onProfileClick = onProfileClick)
+
+    DisplayRecipeCard(
+        presentRecipe = presentRecipe,
+        onLike = { recipeApi.like(presentRecipe) },
+        onDislike = { recipeApi.dislike(presentRecipe) },
+        onReturnClick = onReturnClick,
+        onRecipeClick = onRecipeClick
+    )
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
```

Comment:
I would probably make a `SwipeableStack` composable with a slot `contents: @Composable () -> Unit` that builds children.

Even better, the final signature (with support for stacked items) would look like this :

```
@Composable
fun SwipeableStack<T>(
  items: List<T>,
  modifier: Modifier = Modifier,
  onItemSwiped: ((T) -> Unit)? = {},
  content: @Composable (T) -> Unit,
)
```